mac

```diff
@@ -243,9 +242,18 @@ fancy_echo "patching nodenv init into .zshrc file"
 echo "eval \"\$(nodenv init -)\"" >> .zshrc
 
 # nodenv yarn 
+NODENV_ROOT_PLUGIN="$(nodenv root)/plugins"
+YARN_ROOT="$NODENV_ROOT_PLUGIN/nodenv-yarn-install"
+
 fancy_echo "patching yarn with nodenv"
-mkdir -p "$(nodenv root)/plugins"
-git clone https://github.com/pine/nodenv-yarn-install.git "$(nodenv root)/plugins/nodenv-yarn-install"
+
+if [ ! -d "$YARN_ROOT" ]; then
+  mkdir -p "$NODENV_ROOT_PLUGIN"
+  git clone https://github.com/pine/nodenv-yarn-install.git "$YARN_ROOT"
+else
+  git -C "$YARN_ROOT" pull
+fi
+
```

Comment:
What if `$YARN_ROOT`, for any reason, is not a git repository? Then `git -C "$YARN_ROOT" pull` will fail make the whole script crash. 
If you just want to update the plugin, then you could just remove the folder:

```suggestion
fancy_echo "patching yarn with nodenv"

NODENV_ROOT_PLUGIN="$(nodenv root)/plugins"
if [ ! -d "$NODENV_ROOT_PLUGIN" ]; then
  mkdir -p $NODENV_ROOT_PLUGIN
fi

NODENV_YARN_ROOT="$NODENV_ROOT_PLUGIN/nodenv-yarn-install"
# remove previous plugin installation 
if [ -d "$NODENV_YARN_ROOT" ]; then
  rm -r $NODENV_YARN_ROOT 2> /dev/null
fi

git clone --depth=1 https://github.com/pine/nodenv-yarn-install.git "$NODENV_YARN_ROOT"

```
Does it make sense to you?

Reply:
I personally don't like the idea to remove a someone else's folder, especially if it doesn't contain the .git folder as we expect. I'd prefer to have an exception in this particular case

Reply:
it's a good point!